packages/common/test/location/location_spec.ts

```diff
@@ -215,4 +215,77 @@ describe('Location Class', () => {
       expect(notificationCount).toBe(1);
     });
   });
+
+  describe('location._basePath should contain only base path', () => {
+    const basePath = '/en';
+    const getBaseHref = (origin: string, basePath: string) => origin + basePath + '/';
+
+    it('in case protocol is http:', () => {
+      const origin = 'http://example.com';
+      const baseHref = getBaseHref(origin, basePath);
+
+      TestBed.configureTestingModule({
+        imports: [CommonModule],
+        providers: [
+          {provide: LocationStrategy, useValue: {getBaseHref: () => baseHref, onPopState() {}}},
+          {provide: Location, useClass: Location, deps: [LocationStrategy]},
+        ]
+      });
+
+      const location = TestBed.inject(Location);
+
+      expect((location as any)._basePath).toBe(basePath);
+    });
+
+    it('in case protocol is https:', () => {
+      const origin = 'https://example.com';
+      const baseHref = getBaseHref(origin, basePath);
+
+      TestBed.configureTestingModule({
+        imports: [CommonModule],
+        providers: [
+          {provide: LocationStrategy, useValue: {getBaseHref: () => baseHref, onPopState() {}}},
+          {provide: Location, useClass: Location, deps: [LocationStrategy]},
+        ]
+      });
+
+      const location = TestBed.inject(Location);
+
+      expect((location as any)._basePath).toBe(basePath);
+    });
+
+    it('in case no protocol', () => {
+      const origin = '//example.com';
+      const baseHref = getBaseHref(origin, basePath);
+
+      TestBed.configureTestingModule({
+        imports: [CommonModule],
+        providers: [
+          {provide: LocationStrategy, useValue: {getBaseHref: () => baseHref, onPopState() {}}},
+          {provide: Location, useClass: Location, deps: [LocationStrategy]},
+        ]
+      });
+
+      const location = TestBed.inject(Location);
+
+      expect((location as any)._basePath).toBe(basePath);
+    });
+
+    it('in case no origin', () => {
+      const origin = '';
+      const baseHref = getBaseHref(origin, basePath);
+
+      TestBed.configureTestingModule({
+        imports: [CommonModule],
+        providers: [
+          {provide: LocationStrategy, useValue: {getBaseHref: () => baseHref, onPopState() {}}},
+          {provide: Location, useClass: Location, deps: [LocationStrategy]},
+        ]
```

Comment:
```suggestion
        providers: [{provide: APP_BASE_HREF, useValue: baseHref}]
```
Tests should not mock out more things than necessary because it makes the tests more brittle. In this case, no mocks are necessary.